java/impl/aws/src/main/java/co/worklytics/psoxy/WebhookCollectionModeHandler.java

```diff
@@ -0,0 +1,174 @@
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import co.worklytics.psoxy.aws.DaggerAwsContainer;
+
+/**
+ * AWS lambda entrypoint that can handle BOTH:
+ *    - webhook requests from API Gateway / Function URL invocations, AND
+ *    - SQS events (presumed to be batches of webhooks)
+ */
+public class WebhookCollectionModeHandler implements RequestStreamHandler {
+
+    /**
+     * Static initialization allows reuse in containers
+     * {@link "https://aws.amazon.com/premiumsupport/knowledge-center/lambda-improve-java-function-performance/"}
+     */
+    static AwsContainer awsContainer;
+
+    static BatchMergeHandler batchMergeHandler;
+
+    static InboundWebhookHandler inboundWebhookHandler;
+
+    static ObjectMapper mapper;
+
+    static ObjectMapper sqsPayloadMapper;
+
+    static {
+        staticInit();
+    }
+
+    private static void staticInit() {
+        awsContainer = DaggerAwsContainer.create();
+        batchMergeHandler = awsContainer.batchMergeHandler();
+        inboundWebhookHandler = awsContainer.inboundWebhookHandler();
+        mapper = awsContainer.objectMapper();
+        sqsPayloadMapper = new ObjectMapper();
+        sqsPayloadMapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+    }
+
+
+    @Override
+    public void handleRequest(InputStream input, OutputStream output, Context context) throws IOException {
+        // Read the full input stream into a tree
+        JsonNode rootNode = mapper.readTree(input);
+
+        if (isSQSEvent(rootNode)) {
+            SQSEvent sqsEvent = sqsPayloadMapper.treeToValue(rootNode, SQSEvent.class);
+            handleRequest(sqsEvent, context);
+
+            // Return empty 200 response
+            APIGatewayV2HTTPResponse resp = APIGatewayV2HTTPResponse.builder()
+                .withStatusCode(200)
+                .withBody("SQS event processed")
+                .build();
+            mapper.writeValue(output, resp);
+        } else if (isHttpEvent(rootNode)) {
+            // API Gateway V2 HTTP event (from Function URL, usually)
+            APIGatewayV2HTTPEvent httpEvent = mapper.treeToValue(rootNode, APIGatewayV2HTTPEvent.class);
+            APIGatewayV2HTTPResponse response = handleRequest(httpEvent, context);
+            mapper.writeValue(output, response);
+        } else {
+            context.getLogger().log("Unrecognized event format: " + rootNode.toString());
+            APIGatewayV2HTTPResponse resp = APIGatewayV2HTTPResponse.builder()
+                .withStatusCode(400)
+                .withBody("Unsupported event type")
+                .build();
+            mapper.writeValue(output, resp);
+        }
+    }
+
+    private boolean isSQSEvent(JsonNode node) {
+        return node.has("Records") &&
```

Comment:
What happen if it is a function and the response contains that structure? I think we don't have that in the source, but wondering if we could face an issue here if we compare that structure and not other thing (not sure if we could check headers or other stuff apart from json payload to help identifying this)